### DISCUSSION TITLE
Can we swap Swagger-UI with `redoc`?

### DISCUSSION THREAD

**Discussion by bobmcwhirter** [guacsec/trustify#570] [General]
https://github.com/Redocly/redoc

https://crates.io/crates/utoipa-redoc

Mostly because the Swagger-UI involves too much ugly and entirely too much clicking.

**Comment by bobmcwhirter** [comment]
I'm sure there's more to do with OIDC or such, but duuuuuuuuuude, this is much better.

![image](https://github.com/user-attachments/assets/5f40ab95-17cc-4abd-adf7-015a5021c143)

**Comment by jcrossley3** [comment]
If it allows me to revert https://github.com/trustification/trustify/commit/eba7adb3e97cf685f8b9610dff54345de6caa754 I'm all for it. I hate having to hide that stupid warning.

**Reply by helio-frota** [reply]
:heavy_plus_sign: 
```
2024-05-13T14:04:36.844346Z  WARN actix_web_prom: Cannot build mixed cardinality pattern "/swagger-ui/{_:.*}"
2024-05-13T14:04:36.850008Z  WARN actix_web_prom: Cannot build mixed cardinality pattern "/swagger-ui/{_:.*}"
```

**Comment by bobmcwhirter** [comment]
The downside is we lose the "Try It" functionality, as that's only available in the Redocly product.

**Reply by jcrossley3** [reply]
> The downside is we lose the "Try It" functionality, as that's only available in the Redocly product.

Dude. That's a HUGE downside. I rely on that behavior.

**Comment by bobmcwhirter** [comment]
Another alternative is RapiDoc, which seems to include TryIt.

https://rapidocweb.com/examples/example1.html#post-/pet

https://crates.io/crates/utoipa-rapidoc

**Reply by ctron** [reply]
Added an issue for that: https://github.com/trustification/trustify/issues/578
